app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -57,13 +57,14 @@ public function createOrder($reference, $sessionQuoteId = null, $isPreAuthCreati
 
             $immutableQuoteId = $this->boltHelper()->getImmutableQuoteIdFromTransaction($transaction);
             $immutableQuote = $this->getQuoteById($immutableQuoteId);
+            $immutableQuote->setParentId($transaction->order->cart->order_reference);
             Mage::app()->setCurrentStore($immutableQuote->getStore());
-            $parentQuote = $this->getQuoteById($immutableQuote->getParentQuoteId());
+            $parentQuote = $this->getQuoteById($transaction->order->cart->order_reference);
 
             if (!$parentQuote->getIsActive()) {
                 throw new Exception(
                     $this->boltHelper()->__("The parent quote %s for immutable quote %s is currently being processed or has been processed for order #%s. Check quote %s for details.",
-                        $parentQuote->getId(), $immutableQuote->getId(), $parentQuote->getReservedOrderId(), $parentQuote->getParentQuoteId() )
+                        $transaction->order->cart->order_reference, $immutableQuote->getId(), $parentQuote->getReservedOrderId(), $parentQuote->getParentQuoteId() )
```

Comment:
Is this change necessary since you just redefined `$parentQuote` up there?

Reply:
@matt-thomason You are right.   `$parentQuote->getId()` is useful information, but so is `$transaction->order->cart->order_reference`.  Let's iinclude them both.